packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -155,6 +155,14 @@ export function esbuildDepPlugin(
           const resolved = await resolve(id, importer, kind)
           if (resolved) {
             if (kind === 'require-call') {
+              // #16116 fix: Import the module.scss path, which is actually module.scss.js
+              if (resolved.endsWith('.js')) {
```

Comment:
Is there a reason we only check this in `require-call` only? Wouldn't this help for imports too?

Reply:
I think the reason is because we don't support `import` without extensions in dependencies. But I've noticed that there's a case like `import 'normalize.css'`.

Reply:
Ah right, you have to specify the full path with imports. Let's try this then.